### PR TITLE
callbacks refactoring; bugfixes and plotly for notebooks

### PR DESCRIPTION
**Pull request by Scitator**
- callbacks refactoring, improved 1Cycle
- bugfixes for logging & custom shedulers
- plotly for notebooks

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,19 +1,21 @@
 # Framework must-haves
+numpy>=1.14.6
 torch>=0.4.1
-numpy
+torchvision>=0.2.1
 jpeg4py
 opencv-python
 scikit-image
```
Comment (TezRomacH) [inline requirements.txt:7]: btw, let's freeze any other dependencies?

File: catalyst/utils/plotly.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+from typing import List, Dict, Union, Optional
```
Comment (hexfaker) [inline catalyst/utils/plotly.py:5]: Credits pls

File: examples/_tests_mnist_stages/config2.yml [hunk 3]
```diff
@@ -43,8 +43,8 @@ stages:
 #      scheduler:
 #        callback: OneCycleLR
 #        cycle_len: *num_epochs
-#        div: 3
-#        cut_div: 4
+#        div_factor: 3
```
Comment (hexfaker) [inline examples/_tests_mnist_stages/config2.yml]: What's the reason for leaving commented code? Better remove it anyway

File: catalyst/dl/callbacks/schedulers.py [hunk 4]
```diff
@@ -117,12 +127,14 @@ def calc_lr(self):
             percent = (1 - percent_curr / percent_all)
         else:
             percent = self.cycle_iter / self.cut_point
-        res = self.init_lr * (1 + percent * (self.div - 1)) / self.div
+
+        res = (
+            self.init_lr
```
Comment (hexfaker) [inline catalyst/dl/callbacks/schedulers.py]: Formatting is weird. Mb split into multiple statements?

**Review by hexfaker — CHANGES_REQUESTED** [review]

File: catalyst/utils/plotly.py (hunk 2 above)
Comment (hexfaker) [reply catalyst/utils/plotly.py:5]: To @belskikh

**Review by hexfaker — APPROVED** [review]

**Pull request merged**